src/crypto/crypto_hkdf.cc

```diff
@@ -96,27 +96,57 @@ Maybe<bool> HKDFTraits::AdditionalConfig(
   return Just(true);
 }
 
+char salt[EVP_MAX_KEY_LENGTH];
```

Comment:
```suggestion
char salt[EVP_MAX_KEY_LENGTH] = {0};
```
Otherwise it's uninitialized (i.e., contains whatever garbage is on the stack.)

I suggest moving this closer to where it's used.

Reply:
> I suggest moving this closer to where it's used.

As in inside `HKDFTraits::DeriveBits`? Wouldn't that make it not freed again?

Reply:
Ah, you made it a data member... GH's PR view is so deceptive.

Move it inside the method, to right before the HMAC call that uses it.